lighthouse-core/gather/gatherers/seo/robots-txt.js

```diff
@@ -9,28 +9,29 @@ const Gatherer = require('../gatherer');
 
 /* global fetch, URL, location */
 
+/** @return {Promise<LH.Artifacts['RobotsTxt']>} */
 /* istanbul ignore next */
-function getRobotsTxtContent() {
-  return fetch(new URL('/robots.txt', location.href))
-    .then(response => {
-      if (!response.ok) {
-        return {status: response.status, content: null};
-      }
+async function getRobotsTxtContent() {
+  try {
+    const response = await fetch(new URL('/robots.txt', location.href).href);
+    if (!response.ok) {
+      return {status: response.status, content: null};
+    }
 
-      return response.text()
-        .then(content => ({status: response.status, content}));
-    })
-    .catch(_ => ({status: null, content: null}));
+    return response.text().then(content => ({status: response.status, content}));
```

Comment:
should we just await this too while we're at it? I think we might have to for the semantics to be the same anyhow

Reply:
whoops, good catch